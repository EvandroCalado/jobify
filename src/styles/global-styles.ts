import { createGlobalStyle } from "styled-components";

export const GlobalStyles = createGlobalStyle`
  *,
  *::before,
  *::after {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
  }

  body {
  background: ${({ theme }) => theme.colors.backgroundColor};
  color: ${({ theme }) => theme.colors.textColor};
  font-family: system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-weight: 400;
  line-height: 1;
}

p {
  margin: 0;
}

h1,
h2,
h3,
h4,
h5 {
  margin: 0;
  font-weight: 400;
  line-height: 1;
  text-transform: capitalize;
  letter-spacing: ${({ theme }) => theme.fonts.spacings.letterSpacing};
}

h1 {
  font-size: clamp(2rem, 5vw, 5rem); /* Large heading */
}

h2 {
  font-size: clamp(1.5rem, 3vw, 3rem); /* Medium heading */
}

h3 {
  font-size: clamp(1.25rem, 2.5vw, 2.5rem); /* Small heading */
}

h4 {
  font-size: clamp(1rem, 2vw, 2rem); /* Extra small heading */
}

h5 {
  font-size: clamp(0.875rem, 1.5vw, 1.5rem); /* Tiny heading */
}

.text {
  margin-bottom: 1.5rem;
  max-width: 40em;
}

small,
.text-small {
  font-size: ${({ theme }) => theme.fonts.size.smallText};
}

a {
  text-decoration: none;
}
ul {
  list-style-type: none;
  padding: 0;
}

.img {
  width: 100%;
  display: block;
  object-fit: cover;
}

/* buttons */
.btn {
  cursor: pointer;
  color: ${({ theme }) => theme.colors.white};
  background: ${({ theme }) => theme.colors.primary500};
  border: transparent;
  border-radius: ${({ theme }) => theme.borders.borderRadius};
  letter-spacing: ${({ theme }) => theme.fonts.spacings.letterSpacing};
  padding: 0.65rem;
  box-shadow: ${({ theme }) => theme.shadows.shadow1};
  transition: ${({ theme }) => theme.transitions.fast};
  text-transform: capitalize;
  display: inline-block;
}

.btn:hover {
  background: ${({ theme }) => theme.colors.primary700};
  box-shadow: ${({ theme }) => theme.shadows.shadow3};
}

.btn-hipster {
  color: ${({ theme }) => theme.colors.primary500};
  background: ${({ theme }) => theme.colors.primary200};
}

.btn-hipster:hover {
  color: ${({ theme }) => theme.colors.primary200};
  background: ${({ theme }) => theme.colors.primary700};
}

.btn-block {
  width: 100%;
}

button:disabled {
  cursor: wait;
}

.danger-btn {
  color: ${({ theme }) => theme.colors.redDark};
  background: ${({ theme }) => theme.colors.redLight};
}

.danger-btn:hover {
  color: ${({ theme }) => theme.colors.white};
  background: ${({ theme }) => theme.colors.redDark};
}

/* alerts */
.alert {
  padding: 0.375rem 0.75rem;
  margin-bottom: 1rem;
  border-color: transparent;
  border-radius: ${({ theme }) => theme.borders.borderRadius};
}

.alert-danger {
  color: ${({ theme }) => theme.colors.redDark};
  background: ${({ theme }) => theme.colors.redLight};
}

.alert-success {
  color: ${({ theme }) => theme.colors.greenDark};
  background: ${({ theme }) => theme.colors.greenLight};
}

/* form */
.form {
  width: 90vw;
  max-width: ${({ theme }) => theme.widths.fixedWidth};
  background: ${({ theme }) => theme.colors.backgroundSecondaryColor};
  border-radius: ${({ theme }) => theme.borders.borderRadius};
  box-shadow: ${({ theme }) => theme.shadows.shadow2};
  padding: 2rem 2.5rem;
  margin: 3rem auto;
}

.form-label {
  display: block;
  font-size: ${({ theme }) => theme.fonts.size.smallText};
  margin-bottom: 0.3rem;
  text-transform: capitalize;
  letter-spacing: ${({ theme }) => theme.fonts.spacings.letterSpacing};
  line-height: 1.5;
}

.form-input,
.form-textarea,
.form-select {
  width: 100%;
  padding: 0.375rem 0.75rem;
  border-radius: ${({ theme }) => theme.borders.borderRadius};
  background: ${({ theme }) => theme.colors.backgroundColor};
  border: 1px solid ${({ theme }) => theme.colors.gray300};
  color: ${({ theme }) => theme.colors.textColor};
}

.form-input,
.form-select,
.form-btn {
  height: 35px;
}

.form-row {
  margin-bottom: 1rem;
}

.form-textarea {
  height: 7rem;
}

::placeholder {
  font-family: inherit;
  color: ${({ theme }) => theme.colors.gray400};
}

.form-alert {
  color: ${({ theme }) => theme.colors.redDark};
  letter-spacing: ${({ theme }) => theme.fonts.spacings.letterSpacing};
  text-transform: capitalize;
}

/* alert */
@keyframes spinner {
  to {
    transform: rotate(360deg);
  }
}

.loading {
  width: 6rem;
  height: 6rem;
  border: 5px solid ${({ theme }) => theme.colors.gray400};
  border-radius: 50%;
  border-top-color: ${({ theme }) => theme.colors.primary500};
  animation: spinner 0.6s linear infinite;
}

/* title */
.title {
  text-align: center;
}

.title-underline {
  background: ${({ theme }) => theme.colors.primary500};
  width: 7rem;
  height: 0.25rem;
  margin: 0 auto;
  margin-top: 1rem;
}

.container {
  width: ${({ theme }) => theme.widths.fluidWidth};
  max-width: ${({ theme }) => theme.widths.maxWidth};
  margin: 0 auto;
}

/* BUTTONS AND BADGES */
.pending {
  background: #fef3c7;
  color: #f59e0b;
}

.interview {
  background: #e0e8f9;
  color: #647acb;
}
.declined {
  background: #ffeeee;
  color: #d66a6a;
}
`;
