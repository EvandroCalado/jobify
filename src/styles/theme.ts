export const lightTheme = {
  colors: {
    primary50: "#e0fcff",
    primary100: "#bef8fd",
    primary200: "#87eaf2",
    primary300: "#54d1db",
    primary400: "#38bec9",
    primary500: "#2cb1bc",
    primary600: "#14919b",
    primary700: "#0e7c86",
    primary800: "#0a6c74",
    primary900: "#044e54",

    gray50: "#f8fafc",
    gray100: "#f1f5f9",
    gray200: "#cbd5e1",
    gray300: "#94a3b8",
    gray400: "#64748b",
    gray500: "#475569",
    gray600: "#334155",
    gray700: "#334155",
    gray800: "#1e293b",
    gray900: "#0f172a",

    black: "#222222",
    white: "#ffffff",
    redLight: "#f8d7da",
    redDark: "#842029",
    greenLight: "#d1e7dd",
    greenDark: "#0f5132",

    darkModeBgColor: "#333333",
    darkModeTextColor: "#f0f0f0",
    darkModeBgSecondaryColor: "#3f3f3f",
    darkModeTextSecondaryColor: "#94a3b8",

    backgroundColor: "#f8fafc",
    textColor: "#0f172a",
    backgroundSecondaryColor: "#ffffff",
    textSecondaryColor: "#475569",
  },

  fonts: {
    size: {
      smallText: "0.875rem",
      extraSmallText: "0.7em",
    },
    spacings: {
      letterSpacing: "1px",
    },
  },

  transitions: {
    fast: "0.3s ease-in-out all",
  },

  borders: {
    borderRadius: " 0.25rem",
  },

  widths: {
    maxWidth: "1120px",
    fixedWidth: "600px",
    fluidWidth: "90vw",
    navHeight: "6rem",
  },

  shadows: {
    shadow1: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
    shadow2:
      "0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)",
    shadow3:
      "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
    shadow4:
      "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
  },
} as const;

export const darkTheme = {
  colors: {
    primary50: "#e0fcff",
    primary100: "#bef8fd",
    primary200: "#87eaf2",
    primary300: "#54d1db",
    primary400: "#38bec9",
    primary500: "#2cb1bc",
    primary600: "#14919b",
    primary700: "#0e7c86",
    primary800: "#0a6c74",
    primary900: "#044e54",

    gray50: "#f8fafc",
    gray100: "#f1f5f9",
    gray200: "#cbd5e1",
    gray300: "#94a3b8",
    gray400: "#64748b",
    gray500: "#475569",
    gray600: "#334155",
    gray700: "#334155",
    gray800: "#1e293b",
    gray900: "#0f172a",

    black: "#222222",
    white: "#ffffff",
    redLight: "#f8d7da",
    redDark: "#842029",
    greenLight: "#d1e7dd",
    greenDark: "#0f5132",

    darkModeBgColor: "#333333",
    darkModeTextColor: "#f0f0f0",
    darkModeBgSecondaryColor: "#3f3f3f",
    darkModeTextSecondaryColor: "#94a3b8",

    backgroundColor: "#333333",
    textColor: "#f0f0f0",
    backgroundSecondaryColor: "#3f3f3f",
    textSecondaryColor: "#94a3b8",
  },

  fonts: {
    size: {
      smallText: "0.875rem",
      extraSmallText: "0.7em",
    },
    spacings: {
      letterSpacing: "1px",
    },
  },

  transitions: {
    fast: "0.3s ease-in-out all",
  },

  borders: {
    borderRadius: " 0.25rem",
  },

  widths: {
    maxWidth: "1120px",
    fixedWidth: "600px",
    fluidWidth: "90vw",
    navHeight: "6rem",
  },

  shadows: {
    shadow1: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
    shadow2:
      "0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)",
    shadow3:
      "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
    shadow4:
      "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
  },
} as const;

export type Theme = typeof lightTheme | typeof darkTheme;
