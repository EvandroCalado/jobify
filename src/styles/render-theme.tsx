import { RenderResult, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "./theme";

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider theme={lightTheme}>{children}</ThemeProvider>);
};
