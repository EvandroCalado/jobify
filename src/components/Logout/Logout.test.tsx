import { fireEvent, screen } from "@testing-library/react";
import { Logout } from ".";
import { renderTheme } from "../../styles/render-theme";

describe("<Logout />", () => {
  beforeEach(() => {
    const { container } = renderTheme(<Logout />);

    expect(container).toMatchSnapshot();
  });

  it("should render default", () => {
    expect(screen.getByRole("button", { name: "Guest" })).toBeInTheDocument();
    expect(screen.getByTestId("dropdown")).toBeInTheDocument();
    expect(screen.getByTestId("dropdown")).toHaveStyle({
      visibility: "hidden",
    });
  });

  it("should render visibility visible", () => {
    const button = screen.getByRole("button", { name: "Guest" });

    fireEvent.click(button);

    expect(screen.getByTestId("dropdown")).toHaveStyle({
      visibility: "visible",
    });
  });
});
