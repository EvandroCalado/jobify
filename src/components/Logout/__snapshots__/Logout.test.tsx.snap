// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Logout /> > should render default component 1`] = `
<div>
  <div
    class="sc-beySPh hzmUMK"
  >
    <button
      class="btn logout-btn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
      Guest
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </button>
    <div
      class="dropdown"
      data-testid="dropdown"
    >
      <button
        class="dropdown-btn"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`<Logout /> > should render visibility visible 1`] = `
<div>
  <div
    class="sc-beySPh hzmUMK"
  >
    <button
      class="btn logout-btn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
      Guest
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </button>
    <div
      class="dropdown"
      data-testid="dropdown"
    >
      <button
        class="dropdown-btn"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
